### ISSUE TITLE
xmr-stak 2.5.0 PR Invalid Results on HD6xxx and AMD driver 14.4

### ISSUE THREAD

**Issue by Wacholek** [fireice-uk/xmr-stak#1937]
I have 4 AMD HD6990. No problem with 2.4.7. 

Windows 7, AMD driver 14.4. AMD config auto. CPU config auto. Algo Monero. 
After switch to 2.5.0. :
```
C:\xmr-stak-2.5.0>xmr-stak.exe

[2018-10-15 10:01:57] : Elevating due to Windows 7 or 8. You need Windows 10 to
use fast memory without UAC elevation.
-------------------------------------------------------------------
xmr-stak 2.5.0 90125127

Brought to you by fireice_uk and psychocrypt under GPLv3.
Based on CPU mining code by wolf9466 (heavily optimized by fireice_uk).
Based on NVIDIA mining code by KlausT and psychocrypt.
Based on OpenCL mining code by wolf9466.

Configurable dev donation level is set to 2.0%

-------------------------------------------------------------------
You can use following keys to display reports:
'h' - hashrate
'r' - results
'c' - connection
-------------------------------------------------------------------
Upcoming xmr-stak-gui is sponsored by:
   #####   ______               ____
 ##     ## | ___ \             /  _ \
#    _    #| |_/ /_   _   ___  | / \/ _   _  _ _  _ _  ___  _ __    ___  _   _
#   |_|   #|    /| | | | / _ \ | |   | | | || '_|| '_|/ _ \| '_ \  / __|| | | |
#         #| |\ \| |_| || (_) || \_/\| |_| || |  | | |  __/| | | || (__ | |_| |
 ##     ## \_| \_|\__, | \___/ \____/ \__,_||_|  |_|  \___||_| |_| \___| \__, |
   #####           __/ |                                                  __/ |
                  |___/   https://ryo-currency.com                       |___/

This currency is a way for us to implement the ideas that we were unable to in
Monero. See https://github.com/fireice-uk/cryptonote-speedup-demo for details.
-------------------------------------------------------------------
[2018-10-15 10:01:57] : Mining coin: monero
[2018-10-15 10:01:57] : NVIDIA: try to load library 'xmrstak_cuda_backend_cuda10
_0'
WARNING: NVIDIA Insufficient driver!
[2018-10-15 10:01:57] : NVIDIA: try to load library 'xmrstak_cuda_backend_cuda9_
2'
WARNING: NVIDIA cannot load backend library: xmrstak_cuda_backend_cuda9_2.dll
WARNING: NVIDIA Insufficient driver!
[2018-10-15 10:01:57] : NVIDIA: try to load library 'xmrstak_cuda_backend'
WARNING: NVIDIA Insufficient driver!
[2018-10-15 10:01:57] : WARNING: backend NVIDIA disabled.
[2018-10-15 10:01:57] : Compiling code and initializing GPUs. This will take a w
hile...
[2018-10-15 10:01:58] : Device 0 work size 8 / 32.
[2018-10-15 10:01:58] : OpenCL device 0 - Precompiled code C:\Users\Miwi\AppData
\Local/.openclcache/50d1b9fecfdb779237982dab825dd3bede1f50345e9a11d43de58f827296
29ad.openclbin not found. Compiling ...
LLVM ERROR: Cannot select: 0x638fdd0: i32 = setcc 0x6396830, 0x63a3c10, 0x639815
0 [ORD=2494] [ID=247]
  0x6396830: i64 = add 0x63a3c10, 0x63b11e0 [ORD=2488] [ID=243]
    0x63a3c10: i64 = add 0x63a8e60, 0x63aa370 [ORD=2487] [ID=240]
      0x63a8e60: i64 = mul 0x63a6d40, 0x63a8550 [ORD=2486] [ID=236]
        0x63a6d40: i64 = zero_extend 0x63b01d0 [ORD=2483] [ID=230]
          0x63b01d0: i32 = add 0x6399a70, 0x63a1cf0 [ORD=2478] [ID=224]
            0x6399a70: i32 = AMDILISD::VEXTRACT 0x63afed0, 0x6399060 [ORD=2477]
[ID=219]
              0x63afed0: v2i32 = AMDILISD::BITCONV 0x63a4610 [ORD=2476] [ID=215]

                0x63a4610: i64 = add 0x63a4110, 0x63a6730 [ORD=2475] [ID=209]
                  0x63a4110: i64 = add 0x63a09e0, 0x638f7c0 [ORD=2474] [ID=202]


                  0x63a6730: i64 = zero_extend 0x63a4410 [ORD=2470] [ID=196]

              0x6399060: i32 = TargetConstant<2> [ORD=2777] [ID=84]
            0x63a1cf0: i32,ch = CopyFromReg 0x60a1a00, 0x63ae9c0 [ORD=2478] [ID=
100]
              0x63ae9c0: i32 = Register %vreg47 [ORD=2478] [ID=9]
        0x63a8550: i64 = zero_extend 0x63a3d10 [ORD=2485] [ID=154]
          0x63a3d10: i32 = AMDILISD::VEXTRACT 0x63b5220, 0x6399060 [ORD=2484] [I
D=142]
            0x63b5220: v2i32 = AMDILISD::BITCONV 0x63b11e0 [ORD=2479] [ID=122]
              0x63b11e0: i64,ch = CopyFromReg 0x60a1a00, 0x63a4710 [ORD=2479] [I
D=101]
                0x63a4710: i64 = Register %vreg40 [ORD=2479] [ID=10]
            0x6399060: i32 = TargetConstant<2> [ORD=2777] [ID=84]
      0x63aa370: i64 = zero_extend 0x63ab380 [ORD=2482] [ID=237]
        0x63ab380: i32 = llvm.AMDIL.mulhi.u32 0x6251460, 0x63af7c0, 0x63b01d0 [O
RD=2481] [ID=231]
          0x6251460: i32 = TargetConstant<3113> [ORD=2469] [ID=5]
          0x63af7c0: i32 = AMDILISD::VEXTRACT 0x63b5220, 0x63b8d60 [ORD=2480] [I
D=141]
            0x63b5220: v2i32 = AMDILISD::BITCONV 0x63b11e0 [ORD=2479] [ID=122]
              0x63b11e0: i64,ch = CopyFromReg 0x60a1a00, 0x63a4710 [ORD=2479] [I
D=101]
                0x63a4710: i64 = Register %vreg40 [ORD=2479] [ID=10]
            0x63b8d60: i32 = TargetConstant<1> [ORD=2776] [ID=82]
          0x63b01d0: i32 = add 0x6399a70, 0x63a1cf0 [ORD=2478] [ID=224]
            0x6399a70: i32 = AMDILISD::VEXTRACT 0x63afed0, 0x6399060 [ORD=2477]
[ID=219]
              0x63afed0: v2i32 = AMDILISD::BITCONV 0x63a4610 [ORD=2476] [ID=215]

                0x63a4610: i64 = add 0x63a4110, 0x63a6730 [ORD=2475] [ID=209]
                  0x63a4110: i64 = add 0x63a09e0, 0x638f7c0 [ORD=2474] [ID=202]


                  0x63a6730: i64 = zero_extend 0x63a4410 [ORD=2470] [ID=196]

              0x6399060: i32 = TargetConstant<2> [ORD=2777] [ID=84]
            0x63a1cf0: i32,ch = CopyFromReg 0x60a1a00, 0x63ae9c0 [ORD=2478] [ID=
100]
              0x63ae9c0: i32 = Register %vreg47 [ORD=2478] [ID=9]
    0x63b11e0: i64,ch = CopyFromReg 0x60a1a00, 0x63a4710 [ORD=2479] [ID=101]
      0x63a4710: i64 = Register %vreg40 [ORD=2479] [ID=10]
  0x63a3c10: i64 = add 0x63a8e60, 0x63aa370 [ORD=2487] [ID=240]
    0x63a8e60: i64 = mul 0x63a6d40, 0x63a8550 [ORD=2486] [ID=236]
      0x63a6d40: i64 = zero_extend 0x63b01d0 [ORD=2483] [ID=230]
        0x63b01d0: i32 = add 0x6399a70, 0x63a1cf0 [ORD=2478] [ID=224]
          0x6399a70: i32 = AMDILISD::VEXTRACT 0x63afed0, 0x6399060 [ORD=2477] [I
D=219]
            0x63afed0: v2i32 = AMDILISD::BITCONV 0x63a4610 [ORD=2476] [ID=215]
              0x63a4610: i64 = add 0x63a4110, 0x63a6730 [ORD=2475] [ID=209]
                0x63a4110: i64 = add 0x63a09e0, 0x638f7c0 [ORD=2474] [ID=202]
                  0x63a09e0: i64 = mul 0x63afbd0, 0x63a08e0 [ORD=2473] [ID=197]


                  0x638f7c0: i64 = AMDILISD::LCREATE 0x638f2c0, 0x638edc0 [ORD=2
466] [ID=172]


                0x63a6730: i64 = zero_extend 0x63a4410 [ORD=2470] [ID=196]
                  0x63a4410: i32 = llvm.AMDIL.mulhi.u32 0x6251460, 0x63aeec0, 0x
63a5d30 [ORD=2469] [ID=189]



            0x6399060: i32 = TargetConstant<2> [ORD=2777] [ID=84]
          0x63a1cf0: i32,ch = CopyFromReg 0x60a1a00, 0x63ae9c0 [ORD=2478] [ID=10
0]
            0x63ae9c0: i32 = Register %vreg47 [ORD=2478] [ID=9]
      0x63a8550: i64 = zero_extend 0x63a3d10 [ORD=2485] [ID=154]
        0x63a3d10: i32 = AMDILISD::VEXTRACT 0x63b5220, 0x6399060 [ORD=2484] [ID=
142]
          0x63b5220: v2i32 = AMDILISD::BITCONV 0x63b11e0 [ORD=2479] [ID=122]
            0x63b11e0: i64,ch = CopyFromReg 0x60a1a00, 0x63a4710 [ORD=2479] [ID=
101]
              0x63a4710: i64 = Register %vreg40 [ORD=2479] [ID=10]
          0x6399060: i32 = TargetConstant<2> [ORD=2777] [ID=84]
    0x63aa370: i64 = zero_extend 0x63ab380 [ORD=2482] [ID=237]
      0x63ab380: i32 = llvm.AMDIL.mulhi.u32 0x6251460, 0x63af7c0, 0x63b01d0 [ORD
=2481] [ID=231]
        0x6251460: i32 = TargetConstant<3113> [ORD=2469] [ID=5]
        0x63af7c0: i32 = AMDILISD::VEXTRACT 0x63b5220, 0x63b8d60 [ORD=2480] [ID=
141]
          0x63b5220: v2i32 = AMDILISD::BITCONV 0x63b11e0 [ORD=2479] [ID=122]
            0x63b11e0: i64,ch = CopyFromReg 0x60a1a00, 0x63a4710 [ORD=2479] [ID=
101]
              0x63a4710: i64 = Register %vreg40 [ORD=2479] [ID=10]
          0x63b8d60: i32 = TargetConstant<1> [ORD=2776] [ID=82]
        0x63b01d0: i32 = add 0x6399a70, 0x63a1cf0 [ORD=2478] [ID=224]
          0x6399a70: i32 = AMDILISD::VEXTRACT 0x63afed0, 0x6399060 [ORD=2477] [I
D=219]
            0x63afed0: v2i32 = AMDILISD::BITCONV 0x63a4610 [ORD=2476] [ID=215]
              0x63a4610: i64 = add 0x63a4110, 0x63a6730 [ORD=2475] [ID=209]
                0x63a4110: i64 = add 0x63a09e0, 0x638f7c0 [ORD=2474] [ID=202]
                  0x63a09e0: i64 = mul 0x63afbd0, 0x63a08e0 [ORD=2473] [ID=197]


                  0x638f7c0: i64 = AMDILISD::LCREATE 0x638f2c0, 0x638edc0 [ORD=2
466] [ID=172]


                0x63a6730: i64 = zero_extend 0x63a4410 [ORD=2470] [ID=196]
                  0x63a4410: i32 = llvm.AMDIL.mulhi.u32 0x6251460, 0x63aeec0, 0x
63a5d30 [ORD=2469] [ID=189]



            0x6399060: i32 = TargetConstant<2> [ORD=2777] [ID=84]
          0x63a1cf0: i32,ch = CopyFromReg 0x60a1a00, 0x63ae9c0 [ORD=2478] [ID=10
0]
            0x63ae9c0: i32 = Register %vreg47 [ORD=2478] [ID=9]
In function: __OpenCL_cn111_kernel
`
>

**Comment by psychocrypt** [comment]
we have a few other equal issues. The old driver is not aupoorting one of
the new commands used in the monero pow. I need to figure out which one.
The problem is that I have not such an old system for testing.

**Comment by psychocrypt** [comment]
If you can provide me access via SSH to a Linux System I can debug the issue. Else it will be not possible for me to find out whats wrong. If you are able to give me access please contact me via PM/chat in reddit. (username: psychocrypt)

**Comment by psychocrypt** [comment]
This issue is solved by #1945, never the less please test the PR and report back if all works fine.

**Comment by Wacholek** [comment]
Everything went ok. No problem. 
Only speed is....
i7-3970x from 450h/s on v7 to 385h/s on v8 that is a 14.4% drop. 
HD6990 from 395h/s on v7 to 229h/s on v8 that is a  42% drop per core. 

I do understand that the new algo is less efficient but I think it is to much. 
I played with the intensity and other settings. For me the best where those from the v7.

**Comment by Wacholek** [comment]
Invalid hash rate was fixed on 2.5.1.
